src/main.ts

```diff
@@ -26,12 +27,41 @@ async function run(): Promise<void> {
   try {
     // Get action inputs.
     let projectId = core.getInput('project_id');
+    const cwd = core.getInput('working_directory');
     const deliverables = core.getInput('deliverables');
     const imageUrl = core.getInput('image_url');
     const version = core.getInput('version');
     const promote = core.getInput('promote');
     const serviceAccountKey = core.getInput('credentials');
 
+    // Change working directory
+    if (cwd) {
+      try {
+        process.chdir(cwd.trim());
+      } catch (error) {
+        core.setFailed(error.message);
+      }
+    }
+    // Validate deliverables
+    const allDeliverables = deliverables.split(' ');
+    if (allDeliverables[0] == '') {
+      allDeliverables[0] = 'app.yaml';
+    }
+    let notFound = false;
+    for (const deliverable of allDeliverables) {
+      if (!fs.existsSync(deliverable)) {
+        core.error(deliverable + ' is not in path.');
+        notFound = true;
```

Comment:
Should we fail fast if one of the deliverable is notFound?

Reply:
sgtm